projects/packages/forms/src/blocks/contact-form/class-contact-form-block.php

```diff
@@ -199,8 +199,18 @@ private static function register_beta_blocks() {
 			'jetpack/field-file',
 			array(
 				'render_callback' => array( Contact_Form_Plugin::class, 'gutenblock_render_field_file' ),
+				'plan_check'      => apply_filters( 'jetpack_unauth_file_upload_plan_check', true ),
 			)
 		);
+
+		add_action(
+			'jetpack_register_gutenberg_extensions',
+			function () {
+				if ( ! apply_filters( 'jetpack_unauth_file_upload_plan_check', true ) ) {
+					\Jetpack_Gutenberg::set_extension_available( 'field-file' );
```

Comment:
Instead of having set_extension_available on the Jetpack_Gutenberg here and creating a Jetpack plugin depenency. It would be good to move it to blocks package. Which we already include. 
This way we avoid this dependency.